sandy/decay.py

```diff
@@ -148,6 +156,39 @@ def get_decayconstant(self, with_uncertainty=True):
         else:
             return df.LAMBDA
 
+    def custom_perturbation_decayconstant(self, pert, zam="all"):
+        """
+        Apply a custom perturbation to the decay constant
+        Parameters
+        ----------
+        zam : `int` or `list` or `all`, default is `all`
+            ZAM number of the material to which perturbation is to be
+            applied.
+        pert : 'float'
+            Perturation coeffcient as ratio value.
+        Returns
+        -------
+        `sandy.decay.DecayData`
+            DacayData instance with given value ZAM and decay constant perturbed.
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file("jeff_33",'decay',[922350, 922330, 942390, 942320, 922350])
+        >>> rdd = sandy.DecayData.from_endf6(endf6)
+        >>> tape =sandy.DecayData.custom_perturbation_decayconstant(rdd, 0.05, [922350, 922330])
+        >>> print(tape)
+        922350   3.27689e-17
+        922330   1.44821e-13
+        Name: LAMBDA, dtype: float64
+        """
+        tape = sandy.DecayData.get_decayconstant(self)

```

Comment:
It's weird that you call it `tape`, which is generally used for endf6 files.
This is just a pd.dataframe.

Maybe just call it `dc`.

And since you don't use the uncertainty just do:
`dc = self.get_decayconstant().LAMBDA`.

BTW, I don't get your way of calling self